### PR TITLE
WIP: Batch Discovery Fragment Generator

### PR DESCRIPTION
**Pull request by pongad**

### CODE REVIEW

**Comment by pongad** [comment]
I thought about paralellizing the loop, but 500ms speedup seems to not be worth the extra complexity.

File: src/main/proto/com/google/api/codegen/discovery_batch.proto [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+option java_package = "com.google.api.codegen.discoverybatch";
+option java_outer_classname = "DiscoveryBatchDesc";
+
+message DiscoveryBatch {
+  repeated DiscoverySet include_sets = 1;
+  repeated DiscoverySet ignore_sets = 2;
+  string output = 3;
+}
+
+message DiscoverySet {
```
Comment (garrettjonesgoogle) [inline src/main/proto/com/google/api/codegen/discovery_batch.proto]: Add a comment that this results in a cross product of yaml x disco.

File: src/main/java/com/google/api/codegen/DiscoveryFragmentGeneratorBatch.java [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+      options.set(DiscoveryFragmentGeneratorApi.DISCOVERY_DOC, job.getDiscoPath());
+      options.set(DiscoveryFragmentGeneratorApi.OUTPUT_FILE, job.getOutputDir());
+      options.set(
+          DiscoveryFragmentGeneratorApi.GENERATOR_CONFIG_FILES,
+          ImmutableList.<String>of(job.getYamlPath()));
+      new DiscoveryFragmentGeneratorApi(options).run();
+    }
+  }
+
+  private static Set<Job> makeSet(DiscoverySet discoSet, String outputPath) {
```
Comment (garrettjonesgoogle) [inline src/main/java/com/google/api/codegen/DiscoveryFragmentGeneratorBatch.java]: maybe name this in a way that indicates that a cross-product is being done

Comment (pongad) [inline src/main/java/com/google/api/codegen/DiscoveryFragmentGeneratorBatch.java]: Done.

File: src/main/proto/com/google/api/codegen/discovery_batch.proto (hunk 1 above)
Comment (pongad) [inline src/main/proto/com/google/api/codegen/discovery_batch.proto]: Done.

**Comment by pongad** [comment]
This should be ready for actual review now. @garrettjonesgoogle PTAL.

**Comment by garrettjonesgoogle** [comment]
LGTM

**Pull request merged**